src/components/prompt-text.ts

```diff
@@ -217,10 +207,32 @@ export class PromptText extends EventManager {
                 }
                 case "Word": {
                     if (this.levelData.levelMeta.protoType == "Visible") {
+                    if(this.targetStones.length!=this.currentPromptText.length){
+                    if(this.targetStones.length>i){   
+                    if (this.droppedStoneCount > i || this.droppedStoneCount== undefined) {
+                        this.context.fillStyle = "black";
+                        this.context.fillText(
```

Comment:
This logic can be just used once in this if statement , if the refactored with the above code.